File: dda.py
from matplotlib import pyplot as plt
from random import randint


def DDA(X0, Y0, X1, Y1):
    dx = X1 - X0
    dy = Y1 - Y0

    steps = abs(dx) if abs(dx) > abs(dy) else abs(dy)
    Xinc = float(dx) / float(steps)
    Yinc = float(dy) / float(steps)

    #if((dy/dx)<1):
    for i in range(steps+1):
        x.append(X0)
        y.append(Y0)
        X0+= Xinc
        Y0+= Yinc


for i in range(5):   
    x = []
    y = []
    c, d, e, f = randint(-1000,1000), randint(-1000,1000),randint(-1000,1000),randint(-1000,1000)
    DDA(c,d,e,f)
    print("( {0} , {1} ), ( {2} , {3} )".format(c,d,e,f))
    plt.plot(x,y)
    plt.show()

File: DDA.cpp
/*
By: Pravesh Gaire

*/

#include<iostream>
#include<cmath>

void DDA(int X0, int Y0, int X1, int Y1){
    int dx = X1 - X0;
    int dy = Y1 - Y0;
    
    int steps = abs(dx) > abs(dy) ? abs(dx) : abs(dy);
    float Xinc = dx / (float) steps;
    float Yinc = dy / (float) steps;

    float X = X0;
    float Y = Y0;
    if((dy/dx)>1){
    for (int i=0; i<=steps; i++){
        //putpixel(X,Y,WHITE);
        std::cout<< "(" << X << "," << Y << ")\n";
        X+= Xinc;
        Y+= Yinc;
    }}
    else{
    for(int i=steps; i>=0; i--){
    	std::cout<<"("<<X<<","<<Y<<")\n";
	x-=Xinc;
	y-=Yinc;
    }}
}

int main(int argc, char *argv[]){
    int xa=atoi(argv[1]), xb=atoi(argv[3]), ya=atoi(argv[2]), yb=atoi(argv[4]);
    DDA(xa,ya,xb,yb);
    return 0;
}
